### PR TITLE
Update contribution guideline

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,12 +1,31 @@
 # How to use Jedis Github Issue
 
-* Github issues SHOULD ONLY BE USED to report bugs, and for DETAILED feature requests. Everything else belongs to the Jedis Google Group.
+* Github issues SHOULD ONLY BE USED to report bugs and for DETAILED feature requests. Everything else belongs to the [Jedis Google Group](https://groups.google.com/g/jedis_redis) or [Jedis GitHub Discussions](https://github.com/redis/jedis/discussions).
```
Comment (chayim) [inline .github/CONTRIBUTING.md:3]:
```
Github issues SHOULD USED to report bugs and for DETAILED feature requests. Everything else belongs in the [Jedis Google Group](https://groups.google.com/g/jedis_redis) or [Jedis GitHub Discussions](https://github.com/redis/jedis/discussions).
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 
-Jedis Google Group address:
-  
-https://groups.google.com/forum/?fromgroups#!forum/jedis_redis
+Please post General questions to Google Group or Github discussions. It can be closed without answer when posted to Github issues.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:5]:
```
Please post general questions to Google Group or Github discussions. These will be closed without a response, when posted to GitHub issues.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
 
-Please post General questions to Google Group. It can be closed without answer when posted to Github issue.
+# How to contribute by Pull Request
```
Comment (chayim) [inline .github/CONTRIBUTING.md:7]: # Contributing to Jedis

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
+
+1. Fork Jedis on github (https://help.github.com/articles/fork-a-repo/)
+2. Create a topic branch (git checkout -b my_branch)
+3. Push to your branch (git push origin my_branch)
+4. Post a pull request on github (https://help.github.com/articles/creating-a-pull-request/)
+
+It is recommended to create branch with meaningful name, not modifying master branch directly.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:14]:
```
Create a branch with meaningful name, and do not modify the master branch directly.
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -1,12 +1,31 @@
...
+
+Please add unit tests in order to prove your modification works smoothly. And please make sure your modification passes all unit tests.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:16]:
```
Please add unit tests to validate your changes work, then ensure your changes pass all unit tests.
```

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -1,12 +1,31 @@
...
+
+# Jedis Test Environment
+
+Jedis unit tests run with latest [Redis unstable branch](https://github.com/redis/redis/tree/unstable).
```
Comment (chayim) [inline .github/CONTRIBUTING.md:20]:
```
Jedis unit tests run with the latest [Redis unstable branch](https://github.com/redis/redis/tree/unstable).
Please let them prepared and installed.
```

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -1,12 +1,31 @@
...
+Please let them prepared and installed.
+
+Jedis unit tests use many Redis instances, so we use ```Makefile``` to prepare environment. 
+
+You can start test with ```make test```.
+You can set up test environments by ```make start```, and tear down environments by ```make stop```.
```
Comment (zeekling) [inline .github/CONTRIBUTING.md:26]:
```suggestion
You can set up test environments by ```make start```,  tear down environments by ```make stop```, and clean environments by ``` make cleanup```.
```

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -1,12 +1,31 @@
...
+It is recommended to create branch with meaningful name, not modifying master branch directly.
+
+Please add unit tests in order to prove your modification works smoothly. And please make sure your modification passes all unit tests.
+
+# Jedis Test Environment
+
+Jedis unit tests run with latest [Redis unstable branch](https://github.com/redis/redis/tree/unstable).
+Please let them prepared and installed.
+
+Jedis unit tests use many Redis instances, so we use ```Makefile``` to prepare environment.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:23]: Jedis unit tests use many Redis instances, so we use a```Makefile``` to prepare environment.

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -1,12 +1,31 @@
...
+Please add unit tests in order to prove your modification works smoothly. And please make sure your modification passes all unit tests.
+
+# Jedis Test Environment
+
+Jedis unit tests run with latest [Redis unstable branch](https://github.com/redis/redis/tree/unstable).
+Please let them prepared and installed.
+
+Jedis unit tests use many Redis instances, so we use ```Makefile``` to prepare environment. 
+
+You can start test with ```make test```.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:25]:
```
Start tests with ```make test```.
Set up test environments with ```make start```, and tear down those environments with ```make stop```.
```

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -1,12 +1,31 @@
...
+You can set up test environments by ```make start```, and tear down environments by ```make stop```.
+
+If one or some of unit tests in current master branch of Jedis fails with Redis unstable branch, please post it to Github issue, and go ahead with other unit tests at your work.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:28]:
```
If unit tests fail in the master branch of Jedis fails, while running against the Redis unstable branch, please create a Github issue, and continue testing.
```

**Review by chayim — CHANGES_REQUESTED** [review]
Added some suggestions @sazzad16

**Review by zeekling — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (sazzad16) [reply .github/CONTRIBUTING.md:7]: We consider creating issues as contributions as well. And this block is only about creating PR.

File: .github/CONTRIBUTING.md (hunk 10 above)
Comment (sazzad16) [reply .github/CONTRIBUTING.md]: I have removed the sentence. Our flaky tests could cause a current of issues.

**Review by dengliming — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -1,12 +1,29 @@
 # How to use Jedis Github Issue
 
-* Github issues SHOULD ONLY BE USED to report bugs, and for DETAILED feature requests. Everything else belongs to the Jedis Google Group.
+* Github issues SHOULD BE USED to report bugs and for DETAILED feature requests. Everything else belongs in the [Jedis Google Group](https://groups.google.com/g/jedis_redis) or [Jedis GitHub Discussions](https://github.com/redis/jedis/discussions).
 
-Jedis Google Group address:
-  
-https://groups.google.com/forum/?fromgroups#!forum/jedis_redis
+Please post general questions to Google Group or Github discussions. These can be closed without response when posted to Github issues.
```
Comment (chayim) [inline .github/CONTRIBUTING.md:5]:
```
Please post general questions to Google Groups or Github discussions. These can be closed without response when posted to Github issues.
```

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -1,12 +1,29 @@
...
 
-Please post General questions to Google Group. It can be closed without answer when posted to Github issue.
+# How to contribute by Pull Request
+
+1. Fork Jedis on github (https://help.github.com/articles/fork-a-repo/)
```
Comment (chayim) [inline .github/CONTRIBUTING.md:9]:
```
[Fork](https://help.github.com/articles/fork-a-repo/) Jedis on github
```

Comment (yangbodong22011) [reply .github/CONTRIBUTING.md:9]: Be consistent with the beginning of the article, use `GitHub` instead of `github`

Comment (sazzad16) [reply .github/CONTRIBUTING.md]: @yangbodong22011 Changed the first (and only) `GitHub`

**Review by yangbodong22011 — APPROVED** [review]

**Review by chayim — APPROVED** [review]

**Pull request merged**